### DISCUSSION TITLE
Grid: Using nested grids with nested objects

### DISCUSSION THREAD

**Discussion by hianhianhian** [vikramlearning/blazorbootstrap#835] [Q&A] [answered]
Hi there, I'm trying to use Grids to display a nested object. For example, given the following classes:

```c#
public class Employee
{
    string Name { get; set; }
    List<TaskInfo> TaskList { get; set; } 
}

public class TaskInfo
{
    string Name { get; set; }
}

```

For the above example, it would make sense to display a grid of employees names, and then a nested grid of each employee's list of tasks. How would I actually go about doing this? The example in the docs [here](https://docs.blazorbootstrap.com/components/grid#nested-grid) does not show when using a nested object, because it reuses the same object (which is not a very common use case for nested grids imo)

The problem is that within the `GridDetailView`, how do I access the context object from the outer Grid object?

**Comment by hianhianhian** [comment, marked as answer]
I've resolved this question. The comment from here https://github.com/vikramlearning/blazorbootstrap/discussions/784#discussioncomment-10247459 was the answer, but could be explained a bit better

For the example above, the GridDetailView should look like this:
```razor
<GridDetailView TItem="Employee" Context="employee">
        <Grid TItem="TaskInfo"
              Class="table table-hover border-top"
              DataProvider="(request) => TaskDataProvider(request, employee)">

            <GridColumns>
                <GridColumn TItem="TaskInfo" Context="task" HeaderText="Field" PropertyName="Field">
                    @task.Name
                </GridColumn>
            </GridColumns>
        </Grid>

</GridDetailView>
```

@code
```cs
private async Task<GridDataProviderResult<TaskInfo>> TaskDataProvider(GridDataProviderRequest<TaskInfo> request, Employee employee)
{
    return await Task.FromResult(request.ApplyTo(employee.TaskList));
}
```

**Reply by prizmasoftware** [reply]
Yes, this does the job for displaying the data...
Now there is another problem:
When you try to delete a row from the subgrid list, by deleting it from the database, you need to refresh that subgrid to display it without the deleted row. You will logically use grid.RefreshDataAsync(), then subgrid.RefreshDataAsync(), but this method does not recognize the "addition" you made in the DataProvider and will not refresh the subgrid you changed, only the last one.
Specifically, if you have 3 workers and three subgrids with several items each, if you delete the row and subgrid of the second worker, it will not be refreshed, the subgrid of the third worker will be refreshed, so the refresh will actually work only when you delete rows from the last subgrid .
